test/NuGet.Core.Tests/NuGet.Protocol.Tests/Plugins/PluginDiscovererTests.cs

```diff
@@ -519,6 +519,61 @@ public void GetPluginsInNuGetPluginPaths_NoEnvironmentVariables_ReturnsNoPlugins
             Assert.Empty(plugins);
         }
 
+#if !IS_DESKTOP
+        [PlatformFact(Platform.Windows)]
+        public void GetPluginsInNuGetPluginPaths_Windows_CaseInsensitiveMatching()
+        {
+            // Arrange
+            using var testDirectory = TestDirectory.Create();
+
+            var pluginFilePath1 = Path.Combine(testDirectory.Path, "nuget-plugin-test.exe");
+            var pluginFilePath2 = Path.Combine(testDirectory.Path, "NUGET-PLUGIN-TEST.exe"); // Different casing
+
+            File.Create(pluginFilePath1).Dispose();
+            File.Create(pluginFilePath2).Dispose();
```

Comment:
Since Windows filesystems are, by default, case-insenstive, the second file creation will not work. The test reads like you intended two files to be created, but only one will be wirten, making this look like a test bug.